steam/ext/csgo/client.py

```diff
@@ -58,6 +62,71 @@ async def _handle_ready(self) -> None:
         self.http.user = ClientUser(self._connection, await self.http.get_user(self.user.id64))
         await super()._handle_ready()
 
+    @overload
+    async def inspect_item(self, *, owner: Type[SteamID], asset_id: int, d: int) -> PreviewDataBlock:
+        ...
+
+    @overload
+    async def inspect_item(self, *, s: int, a: int, d: int) -> PreviewDataBlock:
+        ...
+
+    @overload
+    async def inspect_item(self, *, m: int, a: int, d: int) -> PreviewDataBlock:
+        ...
+
+    @overload
+    async def inspect_item(self, *, url: str) -> PreviewDataBlock:
+        ...
+
+    async def inspect_item(
+            self,
+            *,
+            owner: SteamID | None = None,
+            asset_id: int | None = None,
+            s: int = 0,
+            a: int = 0,
+            d: int = 0,
+            m: int = 0,
+            url: str = "",
+            timeout: int=10
+    ) -> PreviewDataBlock:
+        """
+        The parameters can be taken from `inspect` links either from an inventory or market.
+        The market has the `m` parameter, while the inventory one has `s`.
+        :param s: steam id64 of owner
+        :param a: item (asset) id
+        :param d: inspect id (Unknown d number)
+        :param m: market id
+        """
+        if owner:
+            s = owner.id64
+        if asset_id:
+            a = asset_id
+
+        if url:
+            search = re.search(r'[SM](\d+)A(\d+)D(\d+)$', url)
+            s = int(search[1]) if search[0].startswith("S") else 0
+            m = int(search[1]) if search[0].startswith("M") else 0
+            a = int(search[2])
+            d = int(search[3])
+
+        if s == 0 and m == 0:
+            raise TypeError(f"Missing required keyword-only argument: {'s' if not s else 'm'}")
+
+        params = {
+            'param_s': s,
+            'param_a': a,
+            'param_d': d,
+            'param_m': m,
+        }
+        await self.ws.send_gc_message(GCMsgProto(Language.Client2GCEconPreviewDataBlockRequest, **params))
```

Comment:
```suggestion
        await self.ws.send_gc_message(
        	GCMsgProto(
        		Language.Client2GCEconPreviewDataBlockRequest,
        		param_s=s,
        		param_a=a
        		param_d=d,
        		param_m=m,
        	)
        )
```
This reads much better